src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+import {
+  ComponentRegister,
+  LinkComponent,
+  LinkComponentProps
+} from "./ComponentRegister";
+
+const TestLinkComponent: LinkComponent = ({
+  href,
+  children
+}: LinkComponentProps) => <div data-href={href}>{children}</div>;
+
+describe(".init()", () => {
+  it("registers all of the components provided", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    expect(ComponentRegister.Link).toStrictEqual(TestLinkComponent);
+  });
+  it("does not alter properties if none are provided", () => {
+    ComponentRegister.init({});
+
+    expect(ComponentRegister.Link).toStrictEqual("a");
+  });
+});
+it("registers link by default as an anchor tag", () => {
+  expect(
+    <ComponentRegister.Link href="/"> Test</ComponentRegister.Link>
+  ).toStrictEqual(<a href="/"> Test</a>);
+});
+
+describe(".reset()", () => {
+  it("resets ComponentRegister properties back to their default state", () => {
+    ComponentRegister.init({
+      components: {
+        Link: TestLinkComponent
+      }
+    });
+
+    ComponentRegister.reset();
+    expect(ComponentRegister.Link).toStrictEqual("a");
+  });
+});
```

Comment:
Let's also test that the defaults really do match the defaults in the objects. I _think_ that means both without `init` being called, and with `init` called without arguments.

Reply:
Is that not covered by the test that runs init with no arguments? I struggle to see the benfit from init(stuff) -> reset -> check for default & init(empty) -> reset -> check for default

Reply:
Nowhere do we actually test that the thing we export called `default...` actually corresponds to the defaults. That's what I mean to test.